java/TravelSample/src/main/java/com/couchbase/travelsample/model/Hotel.java

```diff
@@ -37,7 +38,7 @@
     public static final String PROP_DESCRIPTION = "description";
 
     @Nonnull
-    public static Hotel fromJSON(@Nonnull JSONObject json) {
+    public static Hotel fromJSON(@Nonnull JSONObject json) throws JSONException {
```

Comment:
No....
JSONException is a subclass of RuntimeException.  That means it is an unchecked exception and does not need to be declared.
again, there's something odd going on here.